src/components/Form/Form.jsx

```diff
@@ -1,20 +1,94 @@
 import React, { useState } from 'react';
 import { useFirebase } from '../../hooks/useFirebase';
+import { useForm } from '../../hooks/useForm';
 
 export const Form = () => {
   const { create } = useFirebase('things');
-  const [thing, setThing] = useState('');
+  const [error, setError] = useState('');
+  const [success, setSuccess] = useState(null);
+
+  const [values, handleInputChange, setValues] = useForm({
+    nameItem: '',
+    selectTime: '',
+    lastDate: null,
+  });
+
   const sendToFB = (e) => {
     e.preventDefault();
-    create({ name: thing });
-    e.target.reset();
+    const itemLength = values.nameItem.length;
```

Comment:
I'm always ultra paranoid about null-pointer errors, which is a risk here because if either `values` or `nameItem` become null, then we'll get a big error. Luckily, JavaScript has a new feature called optional chaining that you can use to avoid null pointers. In the past I had to rely on a log of `&&` or utilities like lodash `get` for this, but now we can just do `values?.nameItem?.length`  

I'm not sure if I've been overusing it in my day to day programming, but I LOVE OPTIONAL CHAINING.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
It has been solved. Optional chaining is really cool, thanks for the tip.